tests/unit/App.spec.ts

```diff
@@ -0,0 +1,121 @@
+
+vi.mock('vue3-gettext', () => ({
+  useGettext: vi.fn().mockImplementation((text) => {
+    return {
+      $gettext: () => text
+    }
+  }),
+  createGettext: vi.fn()
+}))
+
+vi.mock('@cornerstonejs/core', () => {
+  return {
+    successCallback: vi.fn(),
+    errorCallback: vi.fn(),
+    RenderingEngine: class RenderingEngine {
+      getViewport() {
+        return {
+          setStack: vi.fn(),
+          render: vi.fn()
+        }
+      }
+      enableElement() {}
+    },
+    Types: vi.fn(),
+    Enums: {
+      ViewportType: {
+        STACK: '' // "stack",
+      }
+    },
+    metaData: vi.fn(),
+    init: vi.fn(),
+    getConfiguration: vi.fn().mockImplementation(() => {
+      return { rendering: '' }
+    }),
+    registerImageLoader: vi.fn(),
+    isCornerstoneInitialized: vi.fn()
+  }
+})
+
+vi.mock('@cornerstonejs/dicom-image-loader', () => ({
+  wadouri: {
+    loadImage: (imgId) => ({
+      promise: Promise.resolve({})
+    })
+  },
+  external: vi.fn(),
+  configure: vi.fn(),
+  loadImage: vi.fn(),
+  webWorkerManager: {
+    initialize: vi.fn().mockImplementation(() => {})
+  }
+}))
+
+vi.mock('dicom-parser', () => ({
+  default: {}
+}))
+
+describe('App component', () => {
+  describe('Methods', () => {
+    vi.spyOn(App.methods, 'fetchVipMetadataInformation').mockImplementation(vi.fn())
+    vi.spyOn(App.methods, 'fetchMetadataInformation').mockImplementation(vi.fn())
+    const wrapper = getWrapper()
+
+    describe('method: wadouri', () => {
+      it('should add "wadouri" prefix', async () => {
+        expect(await wrapper.vm.addWadouriPrefix('https://dummy_url')).toBe(
+          'wadouri:https://dummy_url'
+        )
+      })
+    })
+
+    describe('method: formatTime', () => {
+      it('should format a valid time', async () => {
+        expect(await wrapper.vm.formatTime('155614.6300')).toBe('15:56:14')
+      })
+      it('should return undefined for invalid time', () => {
+        expect(wrapper.vm.formatTime('')).toBe(undefined)
+        expect(wrapper.vm.formatTime('155')).toBe(undefined)
+      })
+    })
+
+    describe('method: formatDate', () => {
+      it('should format a valid date', () => {
+        expect(wrapper.vm.formatDate('19010101')).toBe('Jan 1, 1901')
+      })
+      it('should return undefined for invalid date', () => {
+        expect(wrapper.vm.formatDate('')).toBe(undefined)
+        expect(wrapper.vm.formatDate('1901010')).toBe(undefined)
+      })
+    })
+
+    describe('method: formatDateAndTime', () => {
+      it('should format a valid date', () => {
+        expect(wrapper.vm.formatDateAndTime('19960823', '093801')).toBe('Aug 23, 1996, 9:38 AM')
+      })
+      it('should return undefined for invalid date or invalid time', () => {
+        expect(wrapper.vm.formatDateAndTime()).toBe(undefined)
+        expect(wrapper.vm.formatDateAndTime('1996082', '093801')).toBe(undefined)
+        expect(wrapper.vm.formatDateAndTime('1996082', '09380')).toBe(undefined)
```

Comment:
they look similar. comments?
```suggestion
        expect(wrapper.vm.formatDateAndTime('1996082', '093801')).toBe(undefined)
        expect(wrapper.vm.formatDateAndTime('1996082', '09380')).toBe(undefined)
```